assets/aml-benchmark/tests/test_dataset_preprocessor.py

```diff
@@ -0,0 +1,283 @@
+        exp_name = f"{self.EXP_NAME}"
+        pipeline_job = self._get_pipeline_job(
+            dataset,
+            template_input,
+            script_path,
+            encoder_config,
+            f"{self.test_dataset_preprocessor_as_component.__name__}-{dataset_name}",
+            pipeline_file="dataset_preprocessor_pipeline.yaml",
+        )
+        pipeline_job = ml_client.create_or_update(pipeline_job, experiment_name=exp_name)
+        ml_client.jobs.stream(pipeline_job.name)
+        print(pipeline_job)
+        self._verify_and_get_output_records(
+            pipeline_job, [dataset], output_dir=os.path.dirname(INPUT_DATASET)
+        )
+        assert_logged_params(
+            pipeline_job.name,
+            exp_name,
+            dataset=[dataset],
+            template_input=template_input,
+            script_path=script_path,
+            encoder_config=encoder_config
+        )
+
+    def _get_pipeline_job(
+        self,
+        dataset: str,
+        template_input: str,
+        script_path: str,
+        encoder_config: str,
+        display_name: str,
+        pipeline_file: str
+    ) -> Job:
+        pipeline_job = load_yaml_pipeline(pipeline_file)
+        # set the pipeline inputs
+        pipeline_job.inputs.dataset = Input(type=AssetTypes.URI_FILE, path=dataset)
+        if template_input:
+            pipeline_job.inputs.template_input = template_input
+        else:
+            pipeline_job.inputs.template_input = None
+        if script_path:
+            pipeline_job.inputs.script_path = Input(type=AssetTypes.URI_FILE, path=script_path)
+        else:
+            pipeline_job.inputs.script_path = None
+        if encoder_config:
+            pipeline_job.inputs.encoder_config = encoder_config
+        else:
+            pipeline_job.inputs.encoder_config = None
+        pipeline_job.display_name = display_name
+        return pipeline_job
+
+    def _verify_and_get_output_records(
+        self,
+        job: Job,
+        input_files: List[str],
+        output_dir: str = None
+    ) -> List[Dict[str, Any]]:
+        """Verify the output and get output records.
+
+        :param job: The pipeline job object.
+        :type job: Job
+        :param output_dir: The local output directory to download pipeline outputs
+        :type output_dir: str
+        :return: output records
+        :rtype: List[Dict[str, Any]]
+        """
+        output_name = job.outputs.output_dataset.port_name
+        if not output_dir:
+            output_dir = Constants.OUTPUT_DIR.format(os.getcwd(), output_name=output_name)
+        else:
+            output_dir = Constants.OUTPUT_DIR.format(output_dir=output_dir, output_name=output_name)
+        download_outputs(
+            job_name=job.name, output_name=output_name, download_path=output_dir
+        )
+        records = _verify_and_get_output_records(input_files, output_dir)
+        return records
+
+
+class TestDatasetPreprocessorScript:

```

Comment:
We are raising so many exception in the scripts, why don't we have tests to validate them? Also, it looks like the tests here are repeat from the component class, we don't want that. The point of this class is to test out different possible values for the input params that could either lead to failure or succeed, which haven't been covered in the component test class.